app/src/main/java/com/example/hepiplant/fragments/PlantsListFragment.java

```diff
@@ -72,10 +101,105 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
+        nameLinearLayout = plantsFragmentView.findViewById(R.id.nameFilterLinearLayout);
+        speciesLinearLayout = plantsFragmentView.findViewById(R.id.speciesFilterLinearLayout);
+        locationLinearLayout = plantsFragmentView.findViewById(R.id.locationFilterLinearLayout);
+    }
+
+    private void setCloseViewsOnClickListeners(){
+        closeNameFilter.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                setRecyclerViewLayoutParams(1);
+                plantNameEditText.setVisibility(View.GONE);
+                nameLinearLayout.setVisibility(View.GONE);
+            }
+        });
+
+        closeSpicesFilter.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                setRecyclerViewLayoutParams(1);
+                speciesSpinner.setVisibility(View.GONE);
+                selectedSpecies=null;
+                speciesLinearLayout.setVisibility(View.GONE);
+            }
+        });
+
+        closeLocationFilter.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                setRecyclerViewLayoutParams(1);
+                locationSpinner.setVisibility(View.GONE);
+                selectedLocation=null;
+                locationLinearLayout.setVisibility(View.GONE);
+            }
+        });
+    }
+
+    private void setRecyclerViewLayoutParams(int plusOrMinus) {
+        ViewGroup.LayoutParams params = plantsRecyclerView.getLayoutParams();
+        params.height =params.height + (plusOrMinus * 130);
+        plantsRecyclerView.setLayoutParams(params);
+    }
+
+    private void setPlantFilterSpinnerAdapter(){
+        plantFilterSpinner.setOnItemSelectedListener(this);
+        ArrayAdapter<String> dtoArrayAdapter = new ArrayAdapter<String>(this.getContext(), android.R.layout.simple_spinner_item, List.of("Filtruj","Nazwa","Gatunek","Pomieszczenie"));
+        dtoArrayAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        plantFilterSpinner.setAdapter(dtoArrayAdapter);
+    }
+
+    private void setPlantFilterButtonOnClickListener() {
+        plantFilterButton.setOnClickListener(new View.OnClickListener() {
+            @RequiresApi(api = Build.VERSION_CODES.N)
+            @Override
+            public void onClick(View v) {
+                if(filterClick%2==0){
+                    makeGetDataRequestWithParam();
+                    plantFilterButton.setText(R.string.clean_button);
+                }
+                else {
+                    makeGetDataRequest();
+                    plantFilterButton.setText(R.string.filter_button);
+                }
+                plantFilterSpinner.setSelection(0);
+                filterClick++;
+            }
+        });
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.N)
+    private void sortPlants(){
```

Comment:
Sortowanie powinno się odbywać na BE, chyba że masz taką funkcjonalność jak w forum sortowanie po dacie. Widzę że u cb zawsze są tak samo te rośliny posortowane więc usuń tą funkcjonalność stąd i dodaj ją proszę na be w endpoincie pobierania wszystkich roślin